File: soknad/src/main/java/no/nav/syfo/kafka/sykepengesoknad/dto/SporsmalDTO.java
package no.nav.syfo.kafka.sykepengesoknad.dto;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.Builder;
import lombok.Value;

import java.util.List;

@Value
@Builder(toBuilder = true)
@JsonIgnoreProperties(ignoreUnknown = true)
public class SporsmalDTO {
    String id;
    String tag;
    String sporsmalstekst;
    String undertekst;
    String min;
    String max;
    Svartype svartype;
    Visningskritere kriterieForVisningAvUndersporsmal;
    List<SvarDTO> svar;
    List<SporsmalDTO> undersporsmal;
}

File: src/main/kotlin/no/nav/helse/flex/sykepengesoknad/kafka/SporsmalDTO.kt
package no.nav.helse.flex.sykepengesoknad.kafka

data class SporsmalDTO(
    val id: String? = null,
    val tag: String? = null,
    val sporsmalstekst: String? = null,
    val undertekst: String? = null,
    val min: String? = null,
    val max: String? = null,
    val svartype: SvartypeDTO? = null,
    val kriterieForVisningAvUndersporsmal: VisningskriteriumDTO? = null,
    val svar: List<SvarDTO>? = null,
    val undersporsmal: List<SporsmalDTO>? = null,
)
